src/test/java/com/synacy/lesson01/exercises/ProductTest.java

```diff
@@ -0,0 +1,24 @@
+package com.synacy.lesson01.exercises;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+/**
+ * Created by michael on 5/3/17.
+ */
+public class ProductTest {
+    @Test
+    public void setUnitPrice_shouldBeAbleToSetPriceGreaterOrEqualToZero() throws Exception {
+        Product product = new Product("sample", 1.0);
+
+        Assert.assertTrue(product.getUnitPrice() >= 0);
+    }
+
+    @Test (expected = IllegalArgumentException.class)
+    public void setUnitPrice_shouldThrowExceptionWhenPriceSetBelowZero() throws Exception {
```

Comment:
Aren't we testing the constructor here based on the method body, are we? It just happens that constructor uses method `setUnitPrice`.